### DISCUSSION TITLE
One or more output arguments not assigned during call to "varargout".

### DISCUSSION THREAD

**Discussion by wendy444444** [yalmip/YALMIP#1122] [General discussion]
Excuse me, why does the program report an error like this? This is my program. I set all the initial values of El to 0, and then changed it to 1 when needed. I don't know why this happened.

**Comment by johanlofberg** [comment]
```
One or more output arguments not
assigned during call to
"varargout".
 
>> El( j, t )

ans =

     0
```

**Reply by wendy444444** [reply]
I don't know what you mean, sir. I set its initial value to 0, and then changed some of them to 1 when needed. Do you mean that it needs to be set for the rest of it?

**Reply by johanlofberg** [reply]
Implies does not support trivial cases like

`sdpvar x;[x <= 10,implies(0,x == 1)]`

Will be fixed, but you can edit implies and change L52 and on to

```
if isempty(X)
    varargout{1} = [];
elseif isa(X,'double') && isequal(X,0)
    varargout{1} = [];
elseif isa(X,'double') && isequal(X,1)
    varargout{1} = Y;
end

```

However, it really make no sense to use implies operator if the first argument is known. If it is a 0, don't add constraint, if it is 1, add it.

**Reply by johanlofberg** [reply]
Fixed in develop branch

**Reply by wendy444444** [reply]
> Implies does not support trivial cases like
> 
> `sdpvar x;[x <= 10,implies(0,x == 1)]`
> 
> Will be fixed, but you can edit implies and change L52 and on to
> 
> ```
> if isempty(X)
>     varargout{1} = [];
> elseif isa(X,'double') && isequal(X,0)
>     varargout{1} = [];
> elseif isa(X,'double') && isequal(X,1)
>     varargout{1} = Y;
> end
> ```
> 
> However, it really make no sense to use implies operator if the first argument is known. If it is a 0, don't add constraint, if it is 1, add it.

Sir, you said L52. Is this the sentence `          C = [C,(implies(Er(i,t),El(Ek(i,1:EE(i)),t)==1)):'El(Ek(i,1:EE(i)),t)==1'` ？Because the type of ER I set is sdpvar and the type of El is zeros, I'm not sure if this is useful for the program.

**Reply by johanlofberg** [reply]
It's implies.m but maybe you'd better run the develop branch instead

**Reply by johanlofberg** [reply]
the code

```
El = zeros
implies(El(1),El(2)==0)
```

which you have makes absolutely no sense. Obviously El is constant and zero, by construction, and impossible to change, so the whole statement is just redundant and states something obviously true

Same thing with

```
implies(Er,El == 1)
```

which makes no sense. El is constant 0.

**Reply by wendy444444** [reply]
Sir, so I can only define the format of sdpvar and then set constraints for it?

**Reply by johanlofberg** [reply]
I have no idea what you are asking
